### PR TITLE
Add bosh acceptance test job

### PR DESCRIPTION
**Pull request by liu-sheng**
Close theopenlab/openlab#4

### CODE REVIEW

**Comment by liu-sheng** [comment]
NOTE: 
- Need to configure the ubuntu-xenial-otc in CI environment firstly
- There are some issues of resource cleanup of this job need to be improve

File: zuul.d/jobs.yaml [hunk 1]
```diff
@@ -332,3 +332,14 @@
     run: playbooks/cf-openstack-validator-acceptance-test/run.yaml
     vars:
       os_branch: 'stable/newton'
+
+- job:
+    name: bosh-acceptance-tests
```
Comment (kiwik) [inline zuul.d/jobs.yaml]: nits: job name and directory name should be s/tests/test/ , keep consistence.

File: playbooks/bosh-acceptance-tests/run.yaml [hunk 2]
```diff
@@ -0,0 +1,137 @@
+- hosts: all
+  become: yes
+  roles:
+    - role: export-cloud-openrc
+      vars:
+        cloud_name: 'opentelekomcloud'
+  tasks:
+    - name: Run cloudfoundry acceptance tests of bosh-acceptance-tests
```
Comment (kiwik) [inline playbooks/bosh-acceptance-tests/run.yaml]: bosh acceptance test is different with cloudfoundry acceptance test, exactly this's should be bosh acceptance tests. So the task name should be "Install bosh and run bosh acceptance tests"

File: playbooks/bosh-acceptance-tests/run.yaml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+          wget --content-disposition https://bosh.io/d/stemcells/bosh-openstack-kvm-ubuntu-trusty-go_agent -O bosh-openstack-kvm-ubuntu-trusty-go_agent.tgz
+          sed -i 's|https://bosh.io/d/stemcells/.*$|file://{{ ansible_user_dir }}/bosh-1/bosh-openstack-kvm-ubuntu-trusty-go_agent.tgz|' bosh-deployment/openstack/cpi.yml
+          sed -i "s/m1.small/${SMALL_FLAVOR}/g" `grep m1.small -rl bosh-deployment/openstack`
+          sed -i "s/m1.xlarge/${LARGE_FLAVOR}/g" `grep m1.xlarge -rl bosh-deployment/openstack`
+          sed -i '/^.*human_readable_vm_names.*$/ a\ \ \ \ state_timeout: 300000' bosh-deployment/openstack/cpi.yml
+
+          apt-get install ruby ruby-dev -y
+
+          _port_id=$(openstack port create --network ${NET_ID} bosh-port -f value -c id)
```
Comment (kiwik) [inline playbooks/bosh-acceptance-tests/run.yaml]: Please add some NOTES about why we create port, then delete it immediately.

File: playbooks/bosh-acceptance-tests/run.yaml [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+          MEDIUM_FLAVOR="s2.large.1" #2U2G
+          LARGE_FLAVOR="s2.2xlarge.1" #8U8G
+          BUILD_ID="{{ zuul.build }}"
+
+          pip install python-openstackclient
+          curl -Lo ./bosh https://s3.amazonaws.com/bosh-cli-artifacts/bosh-cli-3.0.1-linux-amd64
+          chmod +x ./bosh
+          cp ./bosh /usr/local/bin/bosh
+          bosh -v
+          apt-get install -y build-essential zlibc zlib1g-dev ruby ruby-dev openssl libxslt-dev libxml2-dev libssl-dev libreadline6 libreadline6-dev libyaml-dev libsqlite3-dev sqlite3
```
Comment (kiwik) [inline playbooks/bosh-acceptance-tests/run.yaml]: here

File: playbooks/bosh-acceptance-tests/run.yaml [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+          openstack security group rule create --protocol tcp --ingress bosh-${BUILD_ID}
+          openstack security group rule create --protocol icmp --ingress bosh-${BUILD_ID}
+
+          wget --content-disposition https://bosh.io/d/stemcells/bosh-openstack-kvm-ubuntu-trusty-go_agent -O bosh-openstack-kvm-ubuntu-trusty-go_agent.tgz
+          sed -i 's|https://bosh.io/d/stemcells/.*$|file://{{ ansible_user_dir }}/bosh-1/bosh-openstack-kvm-ubuntu-trusty-go_agent.tgz|' bosh-deployment/openstack/cpi.yml
+          sed -i "s/m1.small/${SMALL_FLAVOR}/g" `grep m1.small -rl bosh-deployment/openstack`
+          sed -i "s/m1.xlarge/${LARGE_FLAVOR}/g" `grep m1.xlarge -rl bosh-deployment/openstack`
+          sed -i '/^.*human_readable_vm_names.*$/ a\ \ \ \ state_timeout: 300000' bosh-deployment/openstack/cpi.yml
+
+          apt-get install ruby ruby-dev -y
```
Comment (kiwik) [inline playbooks/bosh-acceptance-tests/run.yaml]: why we install ruby again?

File: playbooks/bosh-acceptance-tests/run.yaml [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+                net_id: ${NET_ID} # Network ID
+                security_groups: [bosh-${BUILD_ID}] # security groups assigned to deployed VMs
+            key_name: microbosh-${BUILD_ID} # (optional) SSH keypair name
+          EOF
+
+          pushd '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}'
+          gem install bundler
+          bundle install
+          cp '{{ ansible_user_dir }}/bosh' /usr/local/bin/bosh
+          bundle exec rspec spec --tag ~vip_networking --tag ~dynamic_networking --tag ~root_partition --tag ~raw_ephemeral_storage
```
Comment (kiwik) [inline playbooks/bosh-acceptance-tests/run.yaml]: why we do not run vip_networking and dynamic_networking test cases?

File: playbooks/bosh-acceptance-tests/run.yaml (hunk 2 above)
Comment (liu-sheng) [reply playbooks/bosh-acceptance-tests/run.yaml]: done

File: zuul.d/jobs.yaml (hunk 1 above)
Comment (liu-sheng) [reply zuul.d/jobs.yaml]: done

File: playbooks/bosh-acceptance-tests/run.yaml (hunk 3 above)
Comment (liu-sheng) [reply playbooks/bosh-acceptance-tests/run.yaml]: done

File: playbooks/bosh-acceptance-tests/run.yaml (hunk 5 above)
Comment (liu-sheng) [reply playbooks/bosh-acceptance-tests/run.yaml]: my mistake :)

File: playbooks/bosh-acceptance-tests/run.yaml (hunk 6 above)
Comment (liu-sheng) [reply playbooks/bosh-acceptance-tests/run.yaml]: just follow the example of the bosh-acceptance-tests, may we should enable all the tests firstly, and to see what tests should be skipped in next step, will change this.

**Comment by liu-sheng** [comment]
- The resource cleanup has been done
- Have fixed the issues according to comments

**Pull request merged**